Lib/asyncio/tasks.py

```diff
@@ -659,6 +672,10 @@ class _GatheringFuture(futures.Future):
     """
 
     def __init__(self, children, *, loop=None):
+        if loop:
```

Comment:
This check is redundant: `asyncio.gather()` always passes `loop` to internal `_GatheringFuture` constructor.